test/memoize.js

```diff
@@ -30,4 +30,25 @@ describe('memoize', function() {
         assert.strictEqual(identity('x'), 'x');
         assert.strictEqual(identity('y'), 'y');
     });
+
+    it('can memoize falsy values', function() {
+        var ctr = 0;
+        var f = R.memoize(function(x) { ctr++; return x; });
+        var values = [0, false, null, undefined, '', NaN];
+        checkFunctionIsIdentityOn(values);
+        // check function has been called once per value
+        assert.equal(values.length, ctr);
+        checkFunctionIsIdentityOn(values);
+        // check function has not been called again for any of the values
+        assert.equal(values.length, ctr);
+        function checkFunctionIsIdentityOn(values) {
+            values.forEach(function(v) {
```

Comment:
Ok, no worries. I've forced pushed to my fork, so that should have updated
in the pull request now.

Peter

On 6 February 2015 at 01:40, Scott Sauyet <notifications@github.com> wrote:

> In test/memoize.js
> <https://github.com/ramda/ramda/pull/776#discussion_r24217285>:
>
> > @@ -30,4 +30,25 @@ describe('memoize', function() {
> >          assert.strictEqual(identity('x'), 'x');
> >          assert.strictEqual(identity('y'), 'y');
> >      });
> > +
> > +    it('can memoize falsy values', function() {
> > +        var ctr = 0;
> > +        var f = R.memoize(function(x) { ctr++; return x; });
> > +        var values = [0, false, null, undefined, '', NaN];
> > +        checkFunctionIsIdentityOn(values);
> > +        // check function has been called once per value
> > +        assert.equal(values.length, ctr);
> > +        checkFunctionIsIdentityOn(values);
> > +        // check function has not been called again for any of the values
> > +        assert.equal(values.length, ctr);
> > +        function checkFunctionIsIdentityOn(values) {
> > +            values.forEach(function(v) {
>
> There has never been a really strong reason to support older engines
> running ES3, especially as ES6 is just around the corner, *except* it has
> generally been fairly easy to do. Since there are still some remote corners
> of the web running on ES3, being able to support them is a nice little bit
> of bragging rights. I would definitely fix this, as it's still pretty
> simple, but at some point, we're going to decide how long we want to keep
> the ES3-compliance.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ramda/ramda/pull/776/files#r24217285>.
>